### PR TITLE
make fracture grid a component (FractureGridGenerator)

### PR DESCRIPTION
**Pull request by gregtucker**
Convert the old fracture_grid utility to a proper component.

### CODE REVIEW

File: landlab/components/fracture_grid/fracture_grid.py [hunk 1]
```diff
@@ -158,55 +138,105 @@ def trace_fracture_through_grid(m, start_yx, spacing):
...
-    m : Numpy array
-        Array containing fracture grid, represented as 0's (matrix) and 1's
-        (fractures). If model_grid parameter is given, returns a 1D array
-        corresponding to a node-based array in the model grid. Otherwise,
-        returns a 2D array with dimensions given by numrows, numcols.
+    Examples
+    --------
+    >>> from landlab import RasterModelGrid
+    >>> grid = RasterModelGrid((5, 5))
+
```
Comment (kbarnhart) [inline landlab/components/fracture_grid/fracture_grid.py]: Should there be an actual example here? Even a really simple one?

File: landlab/components/fracture_grid/fracture_grid.py [hunk 2]
```diff
@@ -158,55 +138,105 @@ def trace_fracture_through_grid(m, start_yx, spacing):
...
+
+    Notes
+    -----
+    Potential improvements:
+    - Add doctests
```
Comment (kbarnhart) [inline landlab/components/fracture_grid/fracture_grid.py]: I think once the example exists this improvement can be removed.

**Comment by gregtucker** [comment]
Oops, forgot to finish doctest. Now done. The .rst files already exist; changed name to component name in one of them. @kbarnhart can you take another look?

**Comment by kbarnhart** [comment]
@gregtucker looks great! When you squash-merge, try using [conventional commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/). I'm still learning how to use them, but I think this would be classified as a `feat`.

**Pull request merged**